meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc

```diff
@@ -0,0 +1,17 @@
+require mender-client.inc
+
+RPROVIDES:${PN} = "mender-client"
+
+DESCRIPTION = "Mender tool for doing OTA software updates."
+HOMEPAGE = "https://mender.io"
+
+SYSTEMD_SERVICE:${PN} = "mender-updated.service mender-authd.service"
```

Comment:
This needs to be split into the package names that contain the services. So it should be:

```
SYSTEMD_SERVICE:mender-auth = "mender-authd.service"
SYSTEMD_SERVICE:mender-update = "mender-updated.service"
```

This goes for quite a lot of our definitions. We used to use `${PN}`, but we will need to start using explicit names.

A slightly unusual thing about this recipe is that I think none of our packages will actually be named "mender-client". I don't know if this is problematic or not, but definitely not the common approach. Usually you have a "main" package with the same name as the recipe, with extra packages on the side.

I asked @TheYoctoJester, and he suggested we name the recipe simply "mender", and then produce the "mender-auth" and "mender-update" packages. Quote:

> if we want the packages `mender-auth` and `mender-update`, then the recipe name should be just `mender`. If the recipe name is `mender-client`, then the packages should be called `mender-client-auth` and `mender-client-update`.

Another possibility would be to completely split the recipe into two recipes with the corresponding names, but this would mean two instances of `SRC_URI` and the various hacks around it.

Reply:
:+1: 

Alright, done the split. Feel free to have another look over